### PR TITLE
Server development

### PR DESCRIPTION
**Pull request by marekrydlewski**
Please ensure it's still working now after refactor :smiling_imp: I had a mysterious merge conflict...

### CODE REVIEW

**Comment by marcin-jablonski** [comment]
IT'S SO NOT WORKING

File: project-ZIP.server/ServerZip.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    t.seekg(0, std::ios::beg);
+    str.assign((std::istreambuf_iterator<char>(t)),
+               std::istreambuf_iterator<char>());
+
+    char *buffer = 0;
+    int length = (int) str.length();
+    buffer = new char[length];
+    std::copy(str.begin(), str.end(), buffer);
+    writeData(_info->connection_fd, sizeof(length), (void *) (&length));
+    writeData(_info->connection_fd, (unsigned int) length, (void *) buffer);
```
Comment (marcin-jablonski) [inline project-ZIP.server/ServerZip.cpp]: Double write?

File: project-ZIP.server/ServerZip.cpp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        if (result < 1) {
+            throw std::length_error(std::string("Error: readXBytes -  wrote less than 1 byte"));
+        }
+
+        bytesWritten += result;
+    }
+}
+
+void ServerZip::writeData(int socket_fd, int x, void *buffer) {
+    writeXBytes(socket_fd, sizeof(x), (void *) &x);
```
Comment (marcin-jablonski) [inline project-ZIP.server/ServerZip.cpp:156]: You can't just add writing something to client without changes in client, you know, it was working before

**Comment by marekrydlewski** [comment]
Check now, I forgot to remove some lines

File: project-ZIP.server/ServerZip.cpp (hunk 2 above)
Comment (marekrydlewski) [inline project-ZIP.server/ServerZip.cpp:156]: Yes, I can - now we use writeData like readData func

**Pull request merged**